onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -444,6 +444,20 @@ struct OpenVINO_Provider : Provider {
       return Status(common::ONNXRUNTIME, ORT_EP_FAIL, "OpenVINO EP only supports one device.");
     }
 
+    // Block setting certain provider options via AppendExecutionProvider_V2
+    const std::unordered_set<std::string> blocked_provider_keys = {
+      "device_type", "device_id", "device_luid", "cache_dir", "precision",
+      "context", "num_of_threads", "model_priority", "num_streams",
+      "enable_opencl_throttling", "enable_qdq_optimizer", "disable_dynamic_shapes"
+    };
+
+    for (const auto& key : blocked_provider_keys) {
+      if (provider_options.find(key) != provider_options.end()) {
+      return Status(common::ONNXRUNTIME, ORT_INVALID_ARGUMENT,
+              "OpenVINO EP: Option '" + key + "' cannot be set explicitly when using AppendExecutionProvider_V2.");
```

Comment:
We may want a separate error messages for keys that we outright block (`device_id", "device_luid", "disable_dynamic_shapes", "context"`), vs. ones that are expected to get passed through `load_config` now.